examples/pub_sub_example.rs

```diff
@@ -0,0 +1,165 @@
+use anyhow::Error;
+use env_logger::Env;
+use std::time::{Duration, Instant};
+use tonari_actor::{Actor, Context, System};
+
+#[derive(Debug)]
+enum PublisherMessage {
+    Periodic,
+    Text(String),
+}
+
+impl From<String> for PublisherMessage {
+    fn from(text: String) -> Self {
+        PublisherMessage::Text(text)
+    }
+}
+
+struct PublisherActor {
+    counter: usize,
+    started_at: Instant,
+}
+
+impl PublisherActor {
+    pub fn new() -> Self {
+        Self { counter: 0, started_at: Instant::now() }
+    }
+}
+
+impl Actor for PublisherActor {
+    type Context = Context<Self::Message>;
+    type Error = Error;
+    type Message = PublisherMessage;
+
+    fn name() -> &'static str {
+        "PublisherActor"
+    }
+
+    fn started(&mut self, context: &mut Self::Context) {
+        context.set_deadline(Some(self.started_at + Duration::from_millis(1500)));
+        context.subscribe::<String>();
+    }
+
+    fn handle(
+        &mut self,
+        context: &mut Self::Context,
+        message: Self::Message,
+    ) -> Result<(), Self::Error> {
+        match message {
+            PublisherMessage::Periodic => {
+                println!(
+                    "Got a periodic message: {:?} at {:?}",
+                    message,
+                    self.started_at.elapsed()
+                );
+
+                let text = format!("Hello from PublisherActor - counter = {}", self.counter);
+                self.counter += 1;
+
+                context.system_handle.publish(text);
+            },
+            PublisherMessage::Text(text) => {
+                println!("PublisherActor got a text message: {:?}", text);
+            },
+        }
+
+        context.system_handle.publish(());
+        Ok(())
+    }
+
+    fn deadline_passed(
+        &mut self,
+        context: &mut Self::Context,
+        deadline: Instant,
+    ) -> Result<(), Error> {
+        context.myself.send(PublisherMessage::Periodic)?;
+        context.set_deadline(Some(deadline + Duration::from_secs(1)));
+        Ok(())
+    }
+}
+
+enum SubscriberMessage {
+    Text(String),
+}
+
+impl From<String> for SubscriberMessage {
```

Comment:
I appreciate using `String` as the "topic" type for simplicity on the example, but I feel like it made it a bit hard to wrap my head around it (It took me a bit to realize that `String` was the topic being subscribed to). 

Maybe a newtype over String would help for the example? `struct Topic(String)`